mlflow/pyspark/ml/__init__.py

```diff
@@ -257,29 +500,101 @@ def _log_pretraining_metadata(estimator, params):
         if params and isinstance(params, dict):
             estimator = estimator.copy(params)
 
-        param_map = _get_instance_param_map(estimator)
-        if isinstance(estimator, Pipeline):
-            pipeline_hierarchy = _get_pipeline_stage_hierarchy(estimator)
+        autologging_metadata = _gen_estimator_metadata(estimator)
+
+        artifact_dict = {}
+
+        param_map = _get_instance_param_map(estimator, autologging_metadata.uid2name_map)
+        if _should_log_hierarchy(estimator):
+            artifact_dict['hierarchy'] = autologging_metadata.hierarchy
+
+        for param_search_estimator in autologging_metadata.param_search_estimators:
+            param_search_estimator_name = \
+                f'{autologging_metadata.uid2name_map[param_search_estimator.uid]}'
+            artifact_dict[param_search_estimator_name] = {}
+
+            artifact_dict[param_search_estimator_name]['tuning_parameter_map_list'] = \
+                _get_tuning_param_maps(
+                    param_search_estimator, autologging_metadata.uid2name_map
+                )
```

Comment:
Can we log a param grid instead of a list of param maps?

```python
# grid: Easier to tell how the search space looks like.
{
  "a.b": [0, 1],
  "c.d": [2, 3],
}

# a list of param maps: Harder to how the search space looks like.
[
  {"a.b": 0, "c.d": 2},
  {"a.b": 0, "c.d": 3},
  {"a.b": 1, "c.d": 2},
  {"a.b": 1, "c.d": 3},
]
```

Reply:
I think we use the `_param_grid` attribute:
https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/python/pyspark/ml/tuning.py#L111

Reply:
but CrossValidator not only support param_grid, but also support tuning on an arbitrary param_map list.

Reply:
Ah got it. If so the current logic is fine.

Reply:
Can we do:

```python
if isinstance(estimatorParamMaps, ParamGridBuilder):
    # log grid
else:
    # log param map list
```

?

Reply:
IMO, either format is fine (though the grid format is certainly more compressed)

Reply:
Actually, if there are 1000s of explored params, a grid is definitely preferable.

Reply:
```
if isinstance(estimatorParamMaps, ParamGridBuilder):
    # log grid
else:
    # log param map list
```
this doesn't work. The estimatorParamMaps will always be a list type, (generated by the ParamGridBuilder.build() )

Reply:
Got it. If so, logging a list of param maps is fine.